pkg/snmp/snmpparse/config_snmp.go

```diff
@@ -97,12 +98,21 @@ func parseConfigSnmpMain(conf config.Component) ([]SNMPConfig, error) {
 		},
 	)
 	//the UnmarshalKey stores the result in mapstructures while the snmpconfig is in yaml
-	//so for each result of the Unmarshal key we storre the result in a tmp SNMPConfig{} object
-	err := conf.UnmarshalKey("snmp_listener.configs", &configs, opt)
-	if err != nil {
-		fmt.Printf("unable to get snmp config from snmp_listener: %v", err)
-		return nil, err
+	//so for each result of the Unmarshal key we store the result in a tmp SNMPConfig{} object
+	if conf.IsSet("network_devices.snmp_listener.configs") {
+		if err := conf.UnmarshalKey("network_devices.snmp_listener.configs", &configs, opt); err != nil {
+			fmt.Printf("unable to get snmp config from network_devices.snmp_listener: %v", err)
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏Not related to this PR as I see `Printf` is used in other places in this file, but I think we should use a log here